src/EntryPoints/REST/ApprovePageApi.php

```diff
@@ -20,30 +19,39 @@ public function __construct(
 	}
 
 	public function run( int $pageId ): Response {
-		// TODO: logic
-		$response = $this->getResponseFactory()->create();
-		$response->setBody( new StringStream( "{ pageId: {$pageId} }" ) );
-
 		$page = $this->getPageIdentity( $pageId );
 
 		if ( $page === null ) {
-			$response->setStatus( 404 );
-			return $response;
+			return $this->presentInvalidPage();
 		}
 
 		if ( !$this->authorizer->canApprove( $page ) ) {
-			$response->setStatus( 403 );
-			return $response;
+			return $this->presentAuthorizationFailed();
 		}
 
-		$response->setStatus( 200 );
-		return $response;
+		// TODO: $persistence->markAsApproved( $pageId, $userId );
+		// TODO: $persistence->saveHtml( $pageId, $html );
+		// TODO: $this->presentApproveFailed();
+
+		return $this->presentSuccess();
 	}
 
 	private function getPageIdentity( int $pageId ): ?PageIdentity {
 		return Title::newFromID( $pageId );
 	}
 
+	public function presentSuccess(): Response {
```

Comment:
Name is a bit weird. Method constructs a response, it does not have a side effect. Would expect something like `newSuccessResponse`